cmd/archeio/app/handlers.go

```diff
@@ -120,38 +123,30 @@ func makeV2Handler(rc RegistryConfig, blobs blobChecker) func(w http.ResponseWri
 		awsRegion, ipIsKnown := regionMapper.GetIP(clientIP)
 		if !ipIsKnown {
 			// no region match, redirect to main upstream registry
-			redirectPath := calculateRedirectPath(rc, path)
-			klog.V(2).InfoS("redirecting blob request to upstream registry", "path", path, "redirect", rc.UpstreamRegistryEndpoint+redirectPath)
-			http.Redirect(w, r, rc.UpstreamRegistryEndpoint+redirectPath, http.StatusTemporaryRedirect)
+			redirectURL := upstreamRedirectURL(rc, rPath)
+			klog.V(2).InfoS("redirecting blob request to upstream registry", "path", rPath, "redirect", redirectURL)
+			http.Redirect(w, r, redirectURL, http.StatusTemporaryRedirect)
 			return
 		}
 
 		// check if blob is available in our S3 bucket for the region
 		bucketURL := awsRegionToS3URL(awsRegion)
-		hash := matches[1]
 		// this matches GCR's GCS layout, which we will use for other buckets
 		blobURL := bucketURL + "/containers/images/sha256%3A" + hash
 		if blobs.BlobExists(blobURL, bucketURL, hash) {
 			// blob known to be available in S3, redirect client there
-			klog.V(2).InfoS("redirecting blob request to S3", "path", path)
+			klog.V(2).InfoS("redirecting blob request to S3", "path", rPath)
 			http.Redirect(w, r, blobURL, http.StatusTemporaryRedirect)
 			return
 		}
 
 		// fall back to redirect to upstream
-		redirectPath := calculateRedirectPath(rc, path)
-		klog.V(2).InfoS("redirecting blob request to upstream registry", "path", path, "redirect", rc.UpstreamRegistryEndpoint+redirectPath)
-		http.Redirect(w, r, rc.UpstreamRegistryEndpoint+redirectPath, http.StatusTemporaryRedirect)
+		redirectURL := upstreamRedirectURL(rc, rPath)
+		klog.V(2).InfoS("redirecting blob request to upstream registry", "path", rPath, "redirect", redirectURL)
+		http.Redirect(w, r, redirectURL, http.StatusTemporaryRedirect)
 	}
 }
 
-func calculateRedirectPath(rc RegistryConfig, path string) string {
```

Comment:
we never call this unless we're redirecting /v2/something to upstream. so we can simplify it a lot.

Reply:
we also always wind up adding the upstream endpoint in front of it, so we can just dedupe that.